### PR TITLE
Call `AuthorizeAndPlayURI` on main thread

### PR DESCRIPTION
**Pull request by haashem**
Fixed a bug caused by my previous PR.

Feel free to change PR title if needed.

Screenshot 2024-09-26 at 20 49 53

### CODE REVIEW

File: ios/spotify_sdk.podspec [hunk 1]
```diff
@@ -15,8 +15,8 @@ Unofficial Spotify Flutter SDK.
   s.source           = { :path => '.' }
   s.source_files = 'Classes/**/*'
   s.dependency 'Flutter'
-  s.platform = :ios, '9.0'
-  s.ios.deployment_target = '9.0'
+  s.platform = :ios, '13.0'
+  s.ios.deployment_target = '13.0'
```
Comment (fotiDim) [inline ios/spotify_sdk.podspec:19]:
```suggestion
  s.platform = :ios, '12.0'
  s.ios.deployment_target = '12.0'
```
In their readme they say "iOS 12 or higher"

Comment (haashem) [reply ios/spotify_sdk.podspec:19]: I had to use `Task` to call their async `authorizeAndPlayURI` method in the sync `connectToSpotify` method, which is only available on iOS 13. 

Screenshot 2024-09-26 at 22 32 17

Comment (fotiDim) [reply ios/spotify_sdk.podspec:19]: It would be great to maintain compatibility with iOS 12. Isn't there another way to call it on the main thread?

Comment (haashem) [reply ios/spotify_sdk.podspec:19]: I doubt they are supporting iOS 12, as you see the method is now using `async` keyword which is only available on iOS 13. 

They are using modern syntax which is not available in iOS 12, so there is no way.

Screenshot 2024-09-27 at 18 21 06

Comment (fotiDim) [reply ios/spotify_sdk.podspec:19]: There are 2 versions of the method. If you use the version with the completion handler you should be able to use it under iOS 12.

Screenshot 2024-09-27 at 21 38 25

Comment (haashem) [reply ios/spotify_sdk.podspec:19]: Good point, but It's not possible to throw error from non-throwable completion block and I don't know a proper way. most Swift cool features are not supported in iOS 12. the last option is to user completion block for `connectToSpotify`.

Screenshot 2024-09-27 at 22 26 06


Supporting iOS 12 is still needed? considering 6 version behind current iOS 18.

Comment (fotiDim) [reply ios/spotify_sdk.podspec:19]: I think we shouldn't be throwing at all here. You may replace throw with `self.connectionStatusHandler?.connectionResult?(FlutterError(code: "spotifyNotInstalled", message: "Spotify app is not installed", details: nil))`.

We match the iOS version support of the native SDK. When the official SDK drops iOS 12 we will drop it as well.

Comment (haashem) [reply ios/spotify_sdk.podspec:19]: I can work on it tomorrow, Thursday.

Comment (brim-borium) [reply ios/spotify_sdk.podspec:19]: Fyi: I fixed the linting issues, just rebase on the current main :). @haashem thanks for contributing to this package!

**Pull request closed without merging**